libs/math/fraction_test.go

```diff
@@ -0,0 +1,68 @@
+package math
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseFraction(t *testing.T) {
+
+	testCases := []struct {
+		inputString string
```

Comment:
I would make these names shorter

f string
exp Fraction
err bool

and flatten the table

Reply:
what do you mean by flatten the table?

Reply:
{X, Y, Z}

instead of 

{
  X,
  Y,
}